### ISSUE TITLE
doubts about the loss function

### ISSUE THREAD

**Issue by LinkToPast1990** [ranandalon/mtl#4]
Does it equal to the multiplication of losses? 
https://piccolboni.info/2018/03/a-simple-loss-function-for-multi-task-learning-with-keras-implementation.html

**Comment by nischnei** [comment]
I think the square in the log is wrong.

**Comment by tsakai-g** [comment]
The formula in the original paper is correct **only when one-dimensional**.

The loss term log(sigma) should have the dimensionality of y as its coefficient. For example, an n-dimensional normal distribution has a coefficient of 1/((sqrt(2*pi)*sigma)**n) so that the integral over the space equals 1, which yields n*log(sigma) in the log-likelihood.

Thanks to the framework providing mean loss functions by default, the balance of the loss terms is correct by chance without taking care of the dimensionality when the method is applied to a problem where the number of pixels is the dimensionality in common.

It is frightening how ML is applied without regard to the principles.

**Comment by AmosDinh** [comment]
@tsakai-g can you tell me why in the code they do 1/(2*sigma**2) ... log(sigma**2)?  Shouldn't it be log(sigma)? 
factor = tf.div(1.0, tf.multiply(2.0, self._sigmas_sq[0]))
loss = tf.add(tf.multiply(factor, self._loss_list[0]), tf.log(self._sigmas_sq[0]))
